course_discovery/apps/api/serializers.py

```diff
@@ -631,8 +631,9 @@ class CourseRunWithProgramsSerializer(CourseRunSerializer):
     programs = serializers.SerializerMethodField()
 
     @classmethod
-    def prefetch_queryset(cls, queryset=None):
+    def prefetch_queryset(cls, queryset=None, edx_org_short_name=None):
         queryset = super().prefetch_queryset(queryset=queryset)
+        queryset = queryset if not edx_org_short_name else queryset.filter(course__authoring_organizations__key=edx_org_short_name)
```

Comment:
We need to create a utils method `get_queryset_filtered_on_organization` so this code should look like:
```python
queryset = super().prefetch_queryset(queryset=queryset)
queryset = get_queryset_filtered_on_organization(queryset, edx_org_short_name)
```

Reply:
Done